hera_cal/apply_cal.py

```diff
@@ -416,8 +429,15 @@ def apply_cal(data_infilename, data_outfilename, new_calibration, old_calibratio
             calibrate_in_place(data, new_gains, data_flags=data_flags, cal_flags=new_flags,
                                old_gains=old_gains, gain_convention=gain_convention)
         if not redundant_average:
-            io.update_vis(data_infilename, data_outfilename, filetype_in=filetype_in, filetype_out=filetype_out,
-                          data=data, flags=data_flags, add_to_history=add_to_history, clobber=clobber, **kwargs)
+            if vis_units is None:
+                if hasattr(hc, 'gain_scale') and hc.gain_scale is not None:
+                    vis_units = hc.gain_scale
+            if vis_units is not None:
+                io.update_vis(data_infilename, data_outfilename, filetype_in=filetype_in, filetype_out=filetype_out,
+                              data=data, flags=data_flags, add_to_history=add_to_history, clobber=clobber, **kwargs, vis_units=vis_units)
+            else:
```

Comment:
I think you can just add vis_units to kwargs, rather than making this if else

Reply:
good catch. Will do!

Reply:
changed this.